packages/indexer-agent/src/agent.ts

```diff
@@ -785,11 +796,31 @@ class Agent {
       poi === null ||
       poi === utils.hexlify(Array(32).fill(0))
     ) {
-      this.logger.error(`Received a null or zero POI for deployment`, {
-        deployment: allocation.subgraphDeployment.id.display,
-        allocation: allocation.id,
-        epochStartBlock,
-      })
+      const indexingStatus = await this.indexer.indexingStatus(allocation.subgraphDeployment.id)
+      const fatalError = indexingStatus.fatalError
+      // can agent check if the error is determinisitic? 
+      // if so inform latest valid POI
```

Comment:
This is a great idea! 
Let's discuss a bit though offline, I believe this is unnecessary with the latest graph node version (just released!) which should always return a POI.